internal/resource/url.go

```diff
@@ -458,25 +469,28 @@ func (f *Fetcher) fetchFromS3(u url.URL, dest s3target, opts FetchOptions) error
 	return nil
 }
 
-func (f *Fetcher) fetchFromS3WithCreds(ctx context.Context, dest s3target, input *s3.GetObjectInput, sess *session.Session) error {
+func (f *Fetcher) fetchFromS3WithCreds(ctx context.Context, dest s3target, input *s3.GetObjectInput, cfg *aws.Config) error {
 	httpClient, err := defaultHTTPClient()
 	if err != nil {
 		return err
 	}
 
-	awsConfig := aws.NewConfig().WithHTTPClient(httpClient)
-	s3Client := s3.New(sess, awsConfig)
-	downloader := s3manager.NewDownloaderWithClient(s3Client)
-	if _, err := downloader.DownloadWithContext(ctx, dest, input); err != nil {
-		if awserrval, ok := err.(awserr.Error); ok && awserrval.Code() == "EC2RoleRequestError" {
-			// If this error was due to an EC2 role request error, try again
-			// with the anonymous credentials.
-			sess.Config.Credentials = credentials.AnonymousCredentials
-			return f.fetchFromS3WithCreds(ctx, dest, input, sess)
+	s3Client := s3.NewFromConfig(*cfg, func(o *s3.Options) {
+		o.HTTPClient = httpClient
+	})
+	downloader := manager.NewDownloader(s3Client)
+	if _, err := downloader.Download(ctx, dest, input); err != nil {
+		// Retry the request as unsigned if error attempting to retrieve/use creds
+		var signingError *v4.SigningError
+		if errors.As(err, &signingError) {
+			_, err = downloader.Download(ctx, dest, input,
+				manager.WithDownloaderClientOptions(func(o *s3.Options) {
+					o.Credentials = aws.AnonymousCredentials{}
+					o.HTTPClient = httpClient
```

Comment:
shouldn't need to override HTTP client again as s3Client already has it overridden

Reply:
done, removed line